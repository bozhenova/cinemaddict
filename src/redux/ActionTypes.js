export const ActionTypes = {
  SET_GENRE: 'SET_GENRE',
  SET_TAB: 'SET_TAB',
  SET_USER_DATA: 'SET_USER_DATA',
  SET_AUTHORIZATION_REQUIRED: 'SET_AUTHORIZATION_REQUIRED',
  SET_ERROR: 'SET_ERROR',
  SET_IS_SENT_STATUS: 'SET_IS_SENT_STATUS',
  SET_IS_SENDING_STATUS: 'SET_IS_SENDING_STATUS',

  LOAD_MOVIES: 'LOAD_MOVIES',
  LOAD_MORE_MOVIES: 'LOAD_MORE_MOVIES',
  LOAD_PROMO_MOVIE: 'LOAD_PROMO_MOVIE',
  LOAD_REVIEWS: 'LOAD_REVIEWS',
  LOAD_FAVORITES: 'LOAD_FAVORITES',

  UPDATE_FAVORITES: 'UPDATE_FAVORITES',

  POST_REVIEW: 'POST_REVIEW'
};
