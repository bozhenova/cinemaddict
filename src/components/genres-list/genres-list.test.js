import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import rootReducer from '../../redux/reducer';
import GenresList from './genres-list.js';

describe('Genres List component renders correctly', () => {
  it('should render a genres list', () => {
    const genres = [];
    const store = createStore(rootReducer);
    const tree = renderer
      .create(
        <Provider store={store}>
          <GenresList genres={genres} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
