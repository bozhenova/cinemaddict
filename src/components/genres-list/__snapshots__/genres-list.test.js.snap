// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genres List component renders correctly should render a genres list 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      data-genre="All genres"
      href="#"
    >
      All genres
    </a>
  </li>
</ul>
`;
